src/com/minseungkwon/datastructure/Graph/BinaryTree.java

```diff
@@ -483,7 +483,50 @@ Result findCommonAncestorUsingPostOrder(Node root, Node n1, Node n2) {
         } else {
             return new Result(rLeft.node != null ? rLeft.node : rRight.node, false);
         }
+    }
+
+    public ArrayList<LinkedList<Node>> findAllCombination(Node root){
+        ArrayList<LinkedList<Node>> results = new ArrayList<>();
+        if(root == null){
+            results.add(new LinkedList<>());
+            return results;
+        }
+        LinkedList<Node> prefix = new LinkedList<>();
+        prefix.add(root);
+        ArrayList<LinkedList<Node>> leftSeq = findAllCombination(root.left);
+        ArrayList<LinkedList<Node>> rightSeq = findAllCombination(root.right);
 
+        for(LinkedList<Node> left : leftSeq){
+            for(LinkedList<Node> right : rightSeq){
+                ArrayList<LinkedList<Node>> weaved = new ArrayList<>();
+                weaveLists(left, right, weaved, prefix);
+                results.addAll(weaved);
+            }
+        }
+        return results;
+    }
+    void weaveLists(LinkedList<Node> left, LinkedList<Node> right, ArrayList<LinkedList<Node>> weaved, LinkedList<Node> prefix){
```

Comment:
Let's make a unit test for this method as well.

Reply:
Okie